bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/HeapSizeManager.java

```diff
@@ -64,60 +74,79 @@ public long getMaxHeapSize() {
   }
 
   public synchronized void waitUntilAllOperationsAreDone() throws InterruptedException {
+    isFlushing.set(true);
     boolean performedWarning = false;
-    while(!pendingOperationsWithSize.isEmpty()) {
+    while (true) {
       cleanupFinishedOperations();
       if (pendingOperationsWithSize.isEmpty()) {
         break;
       }
-      if (!performedWarning
-          && lastOperationChange + INTERVAL_NO_SUCCESS_WARNING < System.currentTimeMillis()) {
-        long lastUpdated = (System.currentTimeMillis() - lastOperationChange) / 1000;
-        LOG.warn("No operations completed within the last %d seconds."
-            + "There are still %d operations in progress.", lastUpdated,
-          pendingOperationsWithSize.size());
+      if (!performedWarning && requiresWarning()) {
+        warn();
         performedWarning = true;
-        waitForCompletions(FINISH_WAIT_MILLIS);
       }
+      wait(FINISH_WAIT_MILLIS);
     }
     if (performedWarning) {
```

Comment:
We have a warning "No operations completed within the last %d seconds."  We wanted to show that only once.